utils/githooks/pre-commit

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# A git hook to validate  and correct the copyright date in source files.
+
+regex='(^\s*\*.*)((Copyright\s*)([0-9]{4})(-([0-9]{4}))?)(\s*(Intel.*$))'
+year=$(date +%Y)
+errors=0
+
+files=$(git diff --diff-filter=AM --name-only --cached -- '*.c' '*.h' 'Makefile')
+for file in $files; do
+	read -r y1 y2 <<< "$(sed -nre "s/^.*$regex.*$/\4 \6/p" "$file")"
```

Comment:
```suggestion
       if [[ "$file" == *vendor* ]] || [[ "$file" == *pb.go ]] || \
               [[ "$file" == *_string.go ]] || [[ "$file" == *pb-c* ]]; then
               continue
       fi

	read -r y1 y2 <<< "$(sed -nre "s/^.*$regex.*$/\4 \6/p" "$file")"
```

This will avoid messing with copyright on generated files.